File: test/tcp.js
var net = require('net');

var HOST = '127.0.0.1';
var PORT = 11111;
var cnt = 1;


var client = new net.Socket();
client.connect(PORT, HOST, function() {
    console.log('CONNECTED TO: ' + HOST + ':' + PORT);
    client.write('I am wayne, from NodeJS !');
});

client.on('data', function(data) {
    console.log('DATA: ' + data);
});

client.on('error', function() {
    console.log('error: ' );
    client.destroy();
});

client.on('close', function() {
    console.log('Connection closed');
});

function intervalFunc() {
	data={"url":"","desc":""};
	tmp = ("0" + cnt++).slice(-2);
	data["url"]="http://www.stevenround-birdphotography.com/source/image/puffin-"
		+ tmp +".jpg";
	data["desc"] = "from node js"+tmp 
	client.write(JSON.stringify(data));
	console.log('send data: '+JSON.stringify(data));
}

setInterval(intervalFunc, 2000);

File: test/tcp.py
import time
import datetime
import json
import socket

address = ('127.0.0.1', 11111)
s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
s.connect(address)

cnt = 0
while True:
    cnt = cnt+1
    name = '{}-{}'.format(datetime.datetime.now(), cnt)
    data = {
        'url': "http://www.stevenround-birdphotography.com/"
                "source/image/puffin-{:02d}.jpg".format(cnt%100),
        'desc': name
    }
    msg = json.dumps(data)
    try:
        s.send(msg.encode())
        print("<---{}".format(msg))
        # time.sleep(1)
        data = s.recv(2048).decode()
        print('--->{}'.format(name, data))
    except KeyboardInterrupt:
        print("canceled by the user")
        break
    except Exception as ex:
        print("Exception:", ex.message)
        # break





s.close()
